src/GafferSceneUI/SelectionTool.cpp

```diff
@@ -229,11 +229,27 @@ bool SelectionTool::buttonPress( const GafferUI::ButtonEvent &event )
 		}
 	}
 
+	m_didRequestDrag = true;
```

Comment:
The name's a bit misleading I think. As I see it, the drag request is made at the point that `dragBegin()` returns non-null, not here in `buttonPress`. Maybe `m_acceptedButtonPress`?

I think we may need to assign `m_acceptedButtonPress = false` at the top of this function too. Consider the following sequence :

1. Button press, `m_acceptedButtonPress = true`.
2. Button release.
3. Button press, we don't accept, but someone else does.
4. Drag begin, we think we accepted the last press, but we didn't.

Actually, even with my suggested fix, I'm still intermittently getting the "selection drag" behaviour if I press-and-hold-v and then click-and-drag. Any ideas?

Reply:
Hopefully fixed in https://github.com/GafferHQ/gaffer/pull/3403/commits/0fff8ffbeccd845baabf516f715a3b5a76a31ee3, along with other edge cases.